WordPress/Classes/Services/ReaderSiteService.m

```diff
@@ -83,6 +83,9 @@ - (void)followSiteWithID:(NSUInteger)siteID success:(void(^)())success failure:(
             return;
         }
         [service followSiteWithID:siteID success:^(){
+            if (success) {
+                success();
+            }
             [WPAnalytics track:WPAnalyticsStatReaderSiteFollowed];
```

Comment:
Nah. At this point the action has happened regardless of whether a success block was passed or not.